src/classes/BDI_ContactService.cls

```diff
@@ -0,0 +1,1095 @@
+            }
+        }
+        if (setHHId.size() > 0) {
+            HH_HouseholdNaming hn = new HH_HouseholdNaming();
+            hn.UpdateNames(new List<Id>(setHHId));
+        }
+    }
+
+    /*******************************************************************************************************
+    * @description for every di with C1 or C2 data, make sure they have a lastname (unless SFId provided).
+    * @return void
+    */
+    private void verifyContactData() {
+        for (DataImport__c di: bdi.listDI) {
+            if (isContactSpecified(di, 'Contact1') && di.Contact1_Lastname__c == null && di.Contact1Imported__c == null) {
+                bdi.LogBDIError(di, label.bdiErrorInvalidLastname, 'Contact1ImportStatus__c');
+                continue;
+            }
+            if (isContactSpecified(di, 'Contact2') && di.Contact2_Lastname__c == null && di.Contact2Imported__c == null) {
+                di.Contact2_Lastname__c = di.Contact1_Lastname__c;
+                if (di.Contact2_Lastname__c == null && di.Contact1Imported__c == null)
+                    bdi.LogBDIError(di, label.bdiErrorInvalidLastname, 'Contact2ImportStatus__c');
+            }
+        }
+    }
+
+    /*******************************************************************************************************
+    * @description looks to see if any of the dikey fields for the Contact are specified.
+    * @param di the Data Import record
+    * @param strCx specifies which contact's fields to look at; either Contact1 or Contact2
+    * @return Boolean true if any of the contact's dikey fields are not empty
+    */
+    private boolean isContactSpecified(DataImport__c di, string strCx) {
+        return di.get(strCx + 'Imported__c') != null ||
+            di.get(strCx + '_Firstname__c') != null ||
+            di.get(strCx + '_Lastname__c') != null ||
+            di.get(strCx + '_Personal_Email__c') != null ||
+            di.get(strCx + '_Work_Email__c') != null ||
+            di.get(strCx + '_Alternate_Email__c') != null ||
+            di.get(strCx + '_Home_Phone__c') != null ||
+            di.get(strCx + '_Mobile_Phone__c') != null ||
+            di.get(strCx + '_Other_Phone__c') != null ||
+            di.get(strCx + '_Work_Phone__c') != null;
+    }
+
+    /*******************************************************************************************************
+    * @description for each DI, updates the appropriate contact with values from the DI and then saves
+    * the Contacts to the database.
+    * @param strCx specifies which contact to update and save; either Contact1 or Contact2
+    * @return Set<Id> the set of DI Id's that were updated
+    */
+    private Set<Id> updateAndSaveContacts(string strCx) {
+        Set<Id> setDIIdUpdated = new Set<Id>();
+        Integer iCx = strCx == 'Contact1' ? 1 : 2;
+
+        // handle multiple di's referring to the same existing contact
+        set<string> setDIKeyUpsert = new set<string>();
```

Comment:
Would you mind capitalizing the first letter of collections and class names? It highlights them differently and makes readability easier.

Reply:
done